test/typescript/middleware.test-d.ts

```diff
@@ -89,6 +89,18 @@ describe('type tests', () => {
     }
   })
 
+  test('middleware has unbound getState function.', () => {
+    type State = { field: 'string' }
+
+    const customMiddleware: Middleware<{}, State> = api => next => action => {
+      const { getState } = api
+
+      expectTypeOf(getState).thisParameter.toEqualTypeOf()
```

Comment:
this doesn't actually fail if the signature is changed back

if we really need a test, then we can switch back to the this: void method, but i'm not convinced that we should worry too much about testing a change solely to satisfy a lint rule, and I think this: void is unnecessary noise for a codebase that doesn't actually use the lint rule it's meant to satisfy

Reply:
I swear I tested this.. thanks for double checking it.

Reply:
that would be my preference - @timdorr?

Reply:
I would add the test as one commit, which should fail, and then add the fix in the next commit to show that this satisfies the condition we're looking to correct.

Reply:
right, but the point is that this is to satisfy a lint rule we're not using - so short of enabling the rule (which i would object to, and would require significant changes to our lint config to enable) I'm not sure how that would work

Reply:
Oh, I see what you're saying. Sorry, I was thinking about the actual test passing. 

As long as the test essentially functions as a lint check, I think that's sufficient. Can we get a spot check by importing this version into a project and seeing if that does pass the lint (vs the current release failing)?

Reply:
I've tested it out in my project and it fixes the lint violation.

Should I remove the test?

Reply:
That seems good enough to me. Leaving the test seems fine, since it at least validates the behavior and will trip if someone breaks this again later on.

Reply:
the test doesn't actually verify anything, and will not break if anything is changed.

Reply:
It's an eslint rule fix. Even if this breaks, it will not cause any bugs in any production environment but only give someone red squigglies. I don't think it needs a test honestly.